src/Aspire.Dashboard/Model/ResourceViewModel.cs

```diff
@@ -324,18 +324,22 @@ public sealed class UrlViewModel
     public string Name { get; }
     public Uri Url { get; }
     public bool IsInternal { get; }
+    public UrlDisplayPropertiesViewModel? DisplayProperties { get; }
 
-    public UrlViewModel(string name, Uri url, bool isInternal)
+    public UrlViewModel(string name, Uri url, bool isInternal, UrlDisplayPropertiesViewModel? displayProperties = null)
     {
         ArgumentException.ThrowIfNullOrWhiteSpace(name);
         ArgumentNullException.ThrowIfNull(url);
 
         Name = name;
         Url = url;
         IsInternal = isInternal;
+        DisplayProperties = displayProperties;
     }
 }
 
+public record UrlDisplayPropertiesViewModel(string? DisplayName, int? SortOrder);
```

Comment:
What happens if displayname is an empty string? I think in that case we treat it the same as null (so ignore the display name and just use the URL).

Reply:
https://github.com/dotnet/aspire/pull/7442#discussion_r1947333828

Reply:
removed nullability